Examples/test-suite/typemap_directorout.i

```diff
@@ -32,6 +32,12 @@
 %typemap(javadirectorout) MyType &Class1::foo2, MyType &foo1 %{ /* special start */ $typemap(javadirectorout, MyType &USEME) /* special end */ %}
 #endif
 
+#ifdef SWIGFORTRAN
+// Since SWIGTYPE& and int& are not compatible in Fortran, we have to replace
+// the typemap
+%typemap(out) MyType &USEME = int &;
```

Comment:
This will need removing if the default array handling becomes the same as the other target languages (under discussion).

Reply:
With most other target languages, the intermediate wrapper for all pointer types (class or fundamental) is a pointer or pointer-like scalar. In the Fortran language module, where memory ownership flags are shared between C++ and the target language, the return value for `SWIGTYPE` is a *struct* that contains a pointer and memory ownership flags, but for non-class objects (such as `int&`) the return value is just a pointer. Since `imtype` and `ctype` and `in` and `out` must all be consistent, swapping out *just* the "out" typemap breaks the Fortran wrapping.